### PR TITLE
Drop twig deferred blocks and deprecated pimcore_* twig extension calls

### PR DESCRIPTION
**Pull request by markus-moser**
Related to 
https://github.com/pimcore/pimcore/pull/17444
https://github.com/pimcore/customer-data-framework/pull/540

* Deferred blocks are not needed, everthing works without the `deferred` attribute too. As this extension does not work with twig >=3.9.0 it's removed now.
  * The only exception could be if someone extends the ecommerce framework twig layouts in custom code and calls the `pimcore_head_link` extension within the content/body block. In that case the call would need to be moved outside of the block. 

* The deprecated usage of `pimcore_head_script().captureStart()` is replaced with an alternative approach.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,6 @@
     "pimcore/google-marketing-bundle": "^1.0",
     "knplabs/knp-paginator-bundle": "^6.0.0",
```
Comment (mattamon) [inline composer.json:22]: Do we need to change the pimcore version too?

Comment (markus-moser) [reply composer.json:22]: Same topic as https://github.com/pimcore/customer-data-framework/pull/540#discussion_r1735934166

Comment (markus-moser) [reply composer.json:22]: No, it's fine without requiring the Pimcore version as we only replaced the depreacted helper with native twig

**Comment by markus-moser** [comment]
As without a updated twig version we have security problems (see https://github.com/pimcore/pimcore/issues/17457#issuecomment-2339887084) we need to remove the deferred blocks extension right now as bugfix as it is incompatible with current versions.

**Pull request merged**